### PR TITLE
Updated containers to preserve capitalization on initialization

### LINKED ISSUE #37: Update `caseInsensitveDict` to preserve initial capitalization
# Description
Currently, keys are cast to lower case on initialization and subsequent addition. We can instead keep those as they are, and just change the string comparison so that they are case insensitive. We should do the same thing for sets also.

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
Closes #37.
The implementation involves storing `self.map` which contains a dictionary for looking up the lower-case and capitalized keys. Any time a new key is added (initialization or subsequently) that capitalization is preserved and cannot be changed, but all comparisons are done with the lower-case values.

Now, for example when printing the options, you get:
```
'+----------------------------------------+'
'|           All test Options:            |'
'+----------------------------------------+'
{'boolOption': True,
 'floatOption': 200.0,
 'intOption': 3,
 'listOption': [],
 'multiOption': {},
 'strOption': 'str1'}
```
where capitalization from initialization is maintained.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
I slightly modified the existing tests.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by marcomangano — COMMENTED** [review]
The addition looks good, I have one major question:
Do we really need `self.map`? Can't we just use `self.invMap = {k.lower(): k for k in self.keys()}`?
It seems like that the "fwd" mapping is not used anywhere else.. and well, it is straight forward as we just turn a string lowercase. The invMap is the only one we actually need as it is not univocal. Anyway, I understand if you want to keep some "symmetry".
Except for that it looks fine!

**Comment by nwu63** [comment]
That is... a really good point. I just assumed I needed both when I wrote this initially but yes the forward mapping is obvious so not needed. I'll push an update.

**Review by marcomangano — APPROVED** [review]

File: baseclasses/utils.py [hunk 1]
```diff
@@ -18,11 +18,9 @@ def __init__(self, *args, **kwargs):
 
     def _updateMaps(self):
```
Comment (sseraj) [inline baseclasses/utils.py:19]: This should probably just be `_updateMap` now.

**Review by sseraj — COMMENTED** [review]
Great suggestion @marcomangano. This is much easier to understand.

**Review by marcomangano — APPROVED** [review]

**Review by sseraj — APPROVED** [review]

**Pull request merged**